docs/xml-sitemaps.md

```diff
@@ -2,70 +2,164 @@
 
-## Indexing content by year
+A sitemap index is automatically linked to in the site's [./robots-txt.md](robots.txt) file at `/wp-sitemap.xml` so you don't need to manually submit them to the [Google Search Console](https://search.google.com/search-console/). You can resubmit them from the search console and get diagnostic information there at any time.
 
-In cases where you have a lot of content (50,000+ pages) you can split your sitemap up into multiple files by year using the following code:
+**Note:** sitemaps are only linked to in the `robots.txt` file is the site is public and in production. To debug locally you will need to define the constant `HM_DISABLE_INDEXING` as `false`.
 
-```php
-add_filter( 'msm_sitemap_index_by_year', '__return_true' );
-```
+## Google Site Verification
 
-## Modifying a sitemap entry
+It is necessary to [verify the site with Google Search Console](https://support.google.com/webmasters/answer/9008080?hl=en) before you can access information about your site's search results performance. It is recommended to use the HTML file upload solution by committing the file to your project's root directory although it is possible to add the meta tag by filling in the [verification code on the Reading Settings page in the admin](admin://options-reading.php).
 
-In order to modify or add extra data to a URL entry in the sitemap XML you can use the `msm_sitemap_entry` filter. Functions attached to this filter can access the current post data using `get_post()` or template tags such as `get_the_permalink()`.
+## Adding Custom Sitemaps
+Sitemaps are provided for built-in content types like pages and author archives out of the box. If you want to add a custom sitemap with additional features such as a video sitemap you can register a custom sitemap provider.
 
-The following example changes the update frequency value and priority for the home page:
+To do so create a custom PHP class that extends the abstract `WP_Sitemaps_Provider` class. Then, you can use the `wp_register_sitemap_provider()` function to register it.
+
+The example below shows a minimal implementation for a custom video sitemap:
 
 ```php
-add_filter( 'msm_sitemap_entry', function ( SimpleXMLElement $url ) : SimpleXMLElement {
-	if ( get_the_permalink() !== get_home_url( '/' ) ) {
-		return $url;
+class Video_Sitemap_Provider extends WP_Sitemaps_Provider {
+
+	/**
+	 * Set the name and object type properties. Required.
+	 */
+	public function __construct() {
+		$this->name = 'videos';
+		$this->object_type = 'post';
 	}
 
-	$url->changefreq = new SimpleXMLElement( '<changefreq>always</changefreq>' );
-	$url->priority = new SimpleXMLElement( '<priority>1.0</priority>' );
+	/**
+	 * Return the list of URLs for the current page.
+	 */
+	public function get_url_list( $page ) : array {
+		$videos = new WP_Query( [
+			'post_type' => 'video',
+			'posts_per_page' => 2000,
```

Comment:
I don't know if this passes our coding standards?

Reply:
It will not no, could add the PHPCS ignore rule in

Reply:
Can we not change it to 100 given it seems to have pagination?

Reply:
Yeah but for a site map it doesn't really make sense, you can have up to 50k sitemaps and each can contain up to 2000 URLs, so why wouldn't you use the max?

Reply:
Because we have limits on the number of posts you can query. I don't remember what our actual limit is though, it may not be 100.

Reply:
The limit only exists in the coding standards, which can be ignored in specific situations, why not this one?

Reply:
```suggestion
			// phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_numberposts
			'posts_per_page' => 2000,
```